### PR TITLE
Jameslotery/dev 2001

### PR DESCRIPTION
**Pull request by jlotery**
Goal of the PR is to migrate from ajv 8.x from 6.x.

Some things needed tweaking such as the use of addKeyword (the actual keyword is now a property of the definition) and not using strict checking on tuples

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (rombat) [inline .vscode/launch.json:1]: is this file necessary ?
Apart from that, lgtm

**Review by rombat — APPROVED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (jlotery) [reply .vscode/launch.json]: good catch, I've deleted it :) thx!
